### PR TITLE
Add event for statement

### PR DESCRIPTION
**Pull request by zhang1002**
### _Why are the changes needed?_
- Through eventLog to write statementInfo into file

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
can we resolve all the conflicts?

**Comment by pan3793** [comment]
Tips: prefer `git rebase` rather than `git merge`

https://dev.to/lydiahallie/cs-visualized-useful-git-commands-37p1
https://zhuanlan.zhihu.com/p/129854679 (same article, Chinese translation)

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/events/StatementEvent.scala [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import org.apache.spark.sql.Encoders
+import org.apache.spark.sql.types.StructType
+
+/**
+ *
+ * @param statementId
+ * @param statement
+ * @param appId
+ * @param sessionId
+ * @param state: store each state that the sql has
```
Comment (yaooqinn) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/events/StatementEvent.scala:29]: Add desc like others

File: externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/events/EventLoggingServiceSuite.scala [hunk 2]
```diff
@@ -83,4 +86,25 @@ class EventLoggingServiceSuite extends WithSparkSQLEngine with JDBCTestUtils {
...
+      req.setSessionHandle(handle)
+      req.setStatement(sql)
+      val tExecuteStatementResp = client.ExecuteStatement(req)
+      val opHandle = tExecuteStatementResp.getOperationHandle
+      val statementId = OperationHandle(opHandle).identifier.toString
+
+      eventually(timeout(60.seconds), interval(5.seconds)) {
+        assert(spark.sql(s"select * from `json`.`${table}`")
+          .where(s"statementId = '${statementId}'")
+          .count() >= 1)
```
Comment (yaooqinn) [inline externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/events/EventLoggingServiceSuite.scala:106]: please also verify other attributes

**Pull request closed without merging**